src/main/kotlin/net/leanix/vsm/sbomBooster/service/ExitScheduler.kt

```diff
@@ -40,7 +40,7 @@ class ExitScheduler(
                 " - Tasks submitted: $tasksSubmitted, Tasks completed: $tasksCompleted," +
                 " Tasks pending: $pendingTasks, Tasks active: $tasksActive"
         )
-        if (pendingTasks == 0L) {
+        if (VsmSbomBoosterApplication.gotRepositories && pendingTasks == 0L) {
```

Comment:
Hi @dsimpson-leanix, 

instead of the introduction of a new variable, we could also increase the `initialDelay` value in L23 to a safe value that will ensure we won't stumble upon the same problem again. 
With the existing settings, we have an initial delay of 30 seconds.  We could increase it to a value of 10 minutes, for example. What do you think?

Reply:
I thought about this as well, but if an org only has 2 repos (or something small), the container wouldn't complete until the ExitScheduler runs (I think). Tracking when it completes getting the repos is the best way to keep it fast but also account for large organizations.

Reply:
If we have only 1–2 repos, we also need to take into account the time the ORT adds to the whole process for cloning, analyzing and generating the SBOM files, which I think in most of the cases would be close to 10 minutes. 

Additionally, the main purpose of SBOM booster is to address the case of organizations with many repos and I acknowledge that we might have some cases that the whole process might run a bit longer than it’s needed, but I think we can afford this.

I am just suggesting to keep complexity as low as possible in order to be able to support faster new features. Don't get me wrong, the implementation is perfectly fine for me, but If you also agree that it's a low priority issue that we foresee to stumble upon in rare cases, maybe we can avoid introducing a new variable.

Reply:
Updated per your recommendation.